rr_common/src/mapper/mapper.cpp

```diff
@@ -89,15 +103,15 @@ int main(int argc, char** argv) {
     while(ros::ok()) {
         ros::spinOnce();
 
-        if(needsUpdating) {
+        if(shouldUpdate()) {
 
             combo_cloud->clear();
             combo_cloud_filtered->clear();
 
-            for(const auto& partial_pair : partials) {
-                // grab the reference to the sensor message from the map
-                auto& msg = partial_pair.second;
-                if(msg->width == 0) continue;
+            for(const auto& entry : queues) {
+                std::queue<sensor_msgs::PointCloud2ConstPtr> queue = entry.second;
+                auto& msg = queue.front();
+                queue.pop();
```

Comment:
msg is a reference to data which no longer guaranteed to exist (bc it has been popped from the container). Not sure how this works